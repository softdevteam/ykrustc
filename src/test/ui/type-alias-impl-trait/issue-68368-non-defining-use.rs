// Regression test for issue #68368
// Ensures that we don't ICE when emitting an error
// for a non-defining use when lifetimes are involved

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete
trait Trait<T> {}
type Alias<'a, U> = impl Trait<U>;
fn f<'a>() -> Alias<'a, ()> {}
//~^ ERROR non-defining opaque type use in defining scope

fn main() {}

impl Trait<()> for () {}
