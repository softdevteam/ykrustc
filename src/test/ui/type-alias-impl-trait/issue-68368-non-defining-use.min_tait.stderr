error: non-defining opaque type use in defining scope
  --> $DIR/issue-68368-non-defining-use.rs:11:15
   |
LL | fn f<'a>() -> Alias<'a, ()> {}
   |               ^^^^^^^^^^^^^
   |
note: used non-generic type `()` for generic parameter
  --> $DIR/issue-68368-non-defining-use.rs:10:16
   |
LL | type Alias<'a, U> = impl Trait<U>;
   |                ^

error: aborting due to previous error

